packages/cli/src/commands/devHandler.js

```diff
@@ -15,6 +16,35 @@ import { getFreePort } from '../lib/ports'
 
 const defaultApiDebugPort = 18911
 
+// @TODO Remove as of Redwood 7.0.0
+const jsDeprecationNotice = () => {
+  if (process.env.REDWOOD_DISABLE_JS_DEPRECATION_NOTICE) {
+    return
+  }
+
+  // There may be actual legitimate JS-only files on the web side, so don't
+  // search ALL files, just the main ones like App, Routes, pages and components
+  const matches = fg.sync(
+    [
+      'web/src/App.js',
+      'web/src/Routes.js',
+      'web/src/components/**/*.js',
+      'web/src/pages/**/*.js',
+    ],
+    { ignore: ['web/src/**/*.mock.js', 'web/src/**/*.routeHooks.js'] }
+  )
+
+  if (matches.length) {
+    console.warn(
+      c.warning(
+        `DEPRECATION NOTICE: When using Vite as your bundler, file extensions for JS files containing JSX must be named \`.jsx\`:\n\n  ${matches.join(
```

Comment:
```suggestion
        `DEPRECATION NOTICE: File extensions for JS files containing JSX must be named \`.jsx\`:\n\n  ${matches.join(
```

because we'll be dropping Webpack in v7 (hopefully!)